### PR TITLE
Cb emp

### LINKED ISSUE #6: Allow developers to access the Employee resource
Verbs to be supported

1. GET
1. POST
1. PUT

User should be able to GET a list, and GET a single item.

### PR DESCRIPTION
**Pull request by CourtneyBrothers**
# Description

Expose employee resource
verbs to be supported 
GET
POST
PUT

Fixes #6 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X ] This change requires a documentation update

# How Has This Been Tested?
1) run nodemon app.js in the project directory

2) to getAllEmployees visit http://localhost:<_your_port number>/api/v1/employees
this should serve the customers JSON 

3) to getOneEmployee  visit http://localhost:<_your_port number>/api/v1/employees/_employee_id

4)postOneEmployee

recommended to use postman to pass employee object 
example: 
{
"firstName": "Afton",
"lastName": "Kessler",
"deptId": 10
}

POST: localhost:<_your_port_number_>/api/v1/employees

employee will be added at next primary integer position to the list of employees

PUT:localhost:<_your_port_number_>/api/v1/employees/_employee_id

employee object will replace the employee object at the _employee_id position specified in the http request 

# Checklist:

- [ ] My code follows the style guidelines of this project
- [X ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X ] I have made corresponding changes to the documentation
- [X ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: models/Employee.js [hunk 1]
```diff
@@ -17,30 +17,30 @@ module.exports.getAll = () => {
...
   });
 }
 
-module.exports.putOne = ({id},{firstName,lastName,addressState,addressStreet,addressZip,addressCity,accountCreationDate}) => {
+module.exports.putOne = ({id},{firstName,lastName,deptId}) => {
   console.log("id",{id});
   return new Promise( (resolve, reject) => {
     
-    db.run(`UPDATE employees SET customer_id = ${id},first_name = "${firstName}",last_name ="${lastName}",address_street="${addressStreet}",address_city = "${addressCity}",address_state = "${addressState}",address_zip = "${addressZip}",account_creation_date = "${accountCreationDate}" WHERE customer_id =${id}`, (err, cust) => {
+    db.run(`UPDATE employees SET employee_id = ${id},first_name = "${firstName}",last_name ="${lastName}",dept_id="${deptId}" WHERE employee_id =${id}`, (err, emp) => {
```
Comment (iiimosley) [inline models/Employee.js]: please remove quotes from `deptId` string interpolation to render value as an integer

File: models/Employee.js [hunk 2]
```diff
@@ -17,30 +17,30 @@ module.exports.getAll = () => {
...
     });
   });
 };
 
 
-module.exports.postOne = ({firstName, lastName, addressState, addressStreet, addressZip, addressCity, accountCreationDate}) => {
+module.exports.postOne = ({firstName, lastName, deptId}) => {
   return new Promise( (resolve, reject) => {
-    db.run(`INSERT INTO employees VALUES(${null},"${firstName}","${lastName}","${addressStreet}","${addressCity}","${addressState}","${addressZip}","${accountCreationDate}")`, (err, cust) => {
+    db.run(`INSERT INTO employees VALUES(${null},"${firstName}","${lastName}","${deptId}")`, (err, emp) => {
```
Comment (iiimosley) [inline models/Employee.js]: please remove quotes from `deptId` string interpolation to render value as an integer

File: models/Employee.js [hunk 3]
```diff
@@ -17,30 +17,30 @@ module.exports.getAll = () => {
 
 module.exports.getOne = (id) => {
   return new Promise( (resolve, reject) => {
-    db.all(`SELECT * FROM employees WHERE customer_id = "${id}"`, (err, cust) => {
+    db.all(`SELECT * FROM employees WHERE employee_id = "${id}"`, (err, emp) => {
```
Comment (iiimosley) [inline models/Employee.js]: please remove quotes from `id` string interpolation to render value as an integer

**Review by iiimosley — CHANGES_REQUESTED** [review]
- Found a few spots in `Employee` model where quotes were inserted around integer values. Please remove so proper integer format may be inserted into the rows.  (see line comments)
- Please reformat your example JSON data in PR to match parameters passed into your `PUT` and `POST` model functions:
  - camelCase rather than snake_case
  - quotes around all keys and string values

Everything else looks great!

**Comment by CourtneyBrothers** [comment]
@iiimosley hey i did this

**Review by iiimosley — APPROVED** [review]

**Review by iiimosley — COMMENTED** [review]
Totes sauce, it has been did.  Thank you!  🍠

**Review by EliOJackson — APPROVED** [review]
I was able to complete all queries 🥇 🎉

**Pull request merged**